TST: stats: `marray` tests failing with latest array-api-strict

e.g.

```
/opt/hostedtoolcache/Python/3.12.11/x64/lib/python3.12/site-packages/array_api_strict/_array_object.py:959: in __setitem__
    other = self._promote_scalar(value)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
        key        = Array(True, dtype=array_api_strict.bool)
        np_key     = array(True)
        other      = False
        self       = Array(0.22652482, dtype=array_api_strict.float64)
        value      = False
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Array(0.22652482, dtype=array_api_strict.float64), scalar = False

    def _promote_scalar(self, scalar: complex) -> Array:
        """
        Returns a promoted version of a Python scalar appropriate for use with
        operations on self.
    
        This may raise an OverflowError in cases where the scalar is an
        integer that is too large to fit in a NumPy integer dtype, or
        TypeError when the scalar type is incompatible with the dtype of self.
        """
        from ._data_type_functions import iinfo
    
        target_dtype = self.dtype
        # Note: Only Python scalar types that match the array dtype are
        # allowed.
        if isinstance(scalar, bool):
            if self.dtype not in _boolean_dtypes:
>               raise TypeError(
                    "Python bool scalars can only be promoted with bool arrays"
E                   TypeError: Python bool scalars can only be promoted with bool arrays

iinfo      = <function iinfo at 0x7f13b953f6a0>
scalar     = False
self       = Array(0.22652482, dtype=array_api_strict.float64)
target_dtype = array_api_strict.float64
```

cc @ev-br @mdhaber

Tried reproducing it locally, but it fails with some other error, both on array-api-strict==2.4.0 and 2.4.1:

```
>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>
> /home/br/repos/scipy/scipy/build-install/usr/lib/python3.13/site-packages/scipy/stats/_stats_py.py(226)gmean()
-> log_a = xp.log(a)
(Pdb) p a
array([MArray(
           array([[0.22652482, 0.43420865, 0.05437862,          _, 0.93584304,
                   0.52332003,          _, 0.73623715],
                  [0.26491467,          _,          _, 0.49263634, 0.41753663,
                   0.84981697, 0.91402933, 0.83559374],
                  [0.09358765,          _,          _, 0.62806941, 0.37649521,
                   0.24324024,          _, 0.89204317],
                  [0.14654981, 0.69961918, 0.58763604, 0.14946943, 0.87875866,
                            _, 0.00903884, 0.62738035],
                  [         _, 0.94150125, 0.1748749 , 0.9313069 , 0.33449384,
                            _, 0.89453604, 0.37493123],
                  [0.54562326, 0.83105279, 0.01028805,          _, 0.96819349,
                   0.79941093, 0.24934093, 0.20682749],
                  [         _, 0.89743012, 0.14070724, 0.6461927 , 0.38779069,
                   0.27299965,          _, 0.43395433]]),
           array([[False, False, False,  True, False, False,  True, False],
                  [False,  True,  True, False, False, False, False, False],
                  [False,  True,  True, False, False, False,  True, False],
                  [False, False, False, False, False,  True, False, False],
                  [ True, False, False, False, False,  True, False, False],
                  [False, False, False,  True, False, False, False, False],
                  [ True, False, False, False, False, False,  True, False]])
       )                                                                      ],
      dtype=object)
(Pdb) q


=============================== short test summary info ===============================
FAILED scipy/stats/tests/test_marray.py::test_xmean[numpy-0-gmean-kwargs0] - TypeError: loop of ufunc does not support argument 0 of type MArray which has no c...

```


Looking at the CI logs, the problem seems to be related to an array-api-strict PR, https://github.com/data-apis/array-api-strict/pull/157, which intended to close https://github.com/data-apis/array-api-strict/issues/136
So from the array-api-strict POV it looks like there's something non-compliant in `marray` or its tests?

--- a/scipy/stats/tests/test_marray.py
+++ b/scipy/stats/tests/test_marray.py
@@ -19,10 +19,6 @@ pytestmark = [
             " is hidden behind SCIPY_ARRAY_API flag."
         ),
     ),
-    pytest.mark.skip_xp_backends(
-        "array_api_strict",
-        reason="scipy/#23412",
-    ),
 ]
 
 skip_backend = pytest.mark.skip_xp_backends

--- a/scipy/stats/tests/test_quantile.py
+++ b/scipy/stats/tests/test_quantile.py
@@ -108,8 +108,7 @@ class TestQuantile:
     @skip_xp_backends(cpu_only=True, reason="PyTorch doesn't have `betainc`.")
     @pytest.mark.parametrize('axis', [0, 1])
     @pytest.mark.parametrize('keepdims', [False, True])
-    # Test with `marray` again when `asarray` supports `device`
-    @pytest.mark.parametrize('nan_policy', ['omit', 'propagate'])  # 'marray'
+    @pytest.mark.parametrize('nan_policy', ['omit', 'propagate', 'marray'])
     @pytest.mark.parametrize('dtype', ['float32', 'float64'])
     @pytest.mark.parametrize('method', ['linear', 'harrell-davis'])
     def test_against_reference(self, axis, keepdims, nan_policy, dtype, method, xp):

